qcodes/instrument_drivers/rohde_schwarz/ZNB.py

```diff
@@ -419,6 +623,79 @@ def _get_sweep_data(self, force_polar: bool = False) -> np.ndarray:
+        self.sweep_type('CW_Point')
+        # turn off average on the VNA since we want single point sweeps.
+        self.write(f'SENS{self._instrument_channel}:AVER:STAT OFF')
+        # This format is required for getting both real and imaginary parts.
+        self.format('Complex')
+        # Set the sweep time to auto such that it sets the delay to zero
+        # between each point (e.g msmt speed is optimized). Note that if one
+        # would like to do a time sweep with time > npts/bandwidth, this is
+        # where the delay would be set, but in general we want to measure as
+        # fast as possible without artificial delays.
+        self.write(f'SENS{self._instrument_channel}:SWE:TIME:AUTO ON')
+        # Set cont measurement off here so we don't have to send that command
+        # while measuring later.
+        self.root_instrument.cont_meas_off()
+
+    def setup_lin_sweep(self) -> None:
+        """
+        Function in order to revert setup_CW_sweep and go back to lin sweep
+        mode.
+        """
+        self.sweep_type('Linear')
+        self.write(f'SENS{self._instrument_channel}:AVER:STAT ON')
+        self.root_instrument.cont_meas_on()
+
+    def _check_cw_sweep(self) -> None:
+        """
+        Checks if all required settings are met to be able to measure in
+        CW_point mode. Similar to what is done in get_sweep_data
+        """
+        if self.sweep_type() != 'CW_Point':
+            raise RuntimeError(f"Sweep type is not set to continuous wave "
+                               f"mode, instead it is: {self.sweep_type()}")
+
+        if not self._parent.rf_power():
+            log.warning("RF output is off when getting sweep data")
+
+        # It is possible that the instrument and QCoDeS disagree about
+        # which parameter is measured on this channel.
+        instrument_parameter = self.vna_parameter()
+        if instrument_parameter != self._vna_parameter:
+            raise RuntimeError("Invalid parameter. Tried to measure "
+                               f"{self._vna_parameter} "
+                               f"got {instrument_parameter}")
+
+        # Turn off average on the VNA since we want single point sweeps.
+        self.write(f'SENS{self._instrument_channel}:AVER:STAT OFF')
+        # Set the format to complex.
+        self.format('Complex')
+        # Set cont measurement off.
+        self.root_instrument.cont_meas_off()
+
+    def _get_cw_data(self, check_cw_sweep_first: bool = True) \
+            -> Tuple[np.ndarray, np.ndarray]:
+        # Make the checking optional such that we can do super fast sweeps as
+        # well, skipping the overhead of the other commands.
+        if check_cw_sweep_first:
+            self._check_cw_sweep()
+
+        with self.status.set_to(1):
+            self.write(f'INIT{self._instrument_channel}:IMM; *WAI')
+            data_str = self.ask(f'CALC{self._instrument_channel}:DATA? SDAT')
```

Comment:
is there a need to wrap this (i don't know which part though, the `*WAI` or `:DATA?``..) into `with self.root_insrtument.timeout.set_to(<smth>):` so that if taking the trace takes more time than the default VISA command timeout, VISA will patiently wait for the measurement and the data transfer to finish before raising a VisaTimeoutError? @LGruenhaupt @abargerbos

Reply:
yeah this could be done, i remember i tried to remove as much unnecessary commands out of the raw get in order to optimize speed of measurements (asking self.sweep_time() is another visa command), since this mode is used for very short measurements(us to ms)

Reply:
@gitjaap so does this mean that it's not possible to set the bandwidth and npts to values that will make the [measurement] + [dataobtaining from the isntrument to PC] time get comparable to the default visa timeout (10 seconds, i think)?

Reply:
Hey its definitely possible, and the instrument will timeout at a time which is the vna.timeout(time), its just that i dont see a reason to change the timeout set by the user.

The reason is I for example sometimes added or forgot a zero in the npts and it suddenly takes a very long time. Then if the timeout is overwritten you also dont timeout before you see the result, isnt that why you have a timeout option?

Reply:
I am on it.